src/main/java/com/keniu/controllers/AuthenticationController.java

```diff
@@ -0,0 +1,41 @@
+package com.keniu.controllers;
+
+import com.keniu.dto.CreateUserRequestDto;
+import com.keniu.dto.UserDto;
+import com.keniu.exceptions.RegistrationException;
+import com.keniu.services.UserService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * Controller class for managing authentication operations in the books shop application.
+ */
+@RequiredArgsConstructor
+@RestController
```

Comment:
according to the task description `POST: /api/auth/registration (User registration)`

```suggestion
@RestController
@RequestMapping("/auth")
```

also, add `server.servlet.context-path=/api` to application.properties

Reply:
OK. Done

Reply:
What about `@RequestMapping("/auth")` from comment above?

Reply:
Sorry, I didn't notice. Added